# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/ScriptLoader.cpp
# Commit: 59207e959b7c
# Full Hash: 59207e959b7c3476cc21409d61c8dd03d22a816b
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-12-21 21:21:23
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement requestCredentials for Module Workers; r=jonco
#   
#   This weaves credentials through from the private to script creation.
#   
#   Depends on D155567
# ==============================================================================

diff -r 49f18430c465 -r 59207e959b7c dom/workers/ScriptLoader.cpp
--- a/dom/workers/ScriptLoader.cpp	Tue Dec 20 20:56:13 2022 +0000
+++ b/dom/workers/ScriptLoader.cpp	Tue Dec 20 20:56:13 2022 +0000
@@ -523,7 +523,7 @@
     loadContext->mLoadResult = rv;
   }
 
-  RefPtr<ScriptLoadRequest> request;
+  RefPtr<ScriptLoadRequest> request = nullptr;
   if (mWorkerRef->Private()->WorkerType() == WorkerType::Classic) {
     RefPtr<ScriptFetchOptions> fetchOptions =
         new ScriptFetchOptions(CORSMode::CORS_NONE, referrerPolicy, nullptr);
@@ -533,12 +533,19 @@
                                     loadContext);
   } else {
     // Implements part of "To fetch a worklet/module worker script graph"
-    // including, setting up the request with a credentials mode (TODO),
+    // including, setting up the request with a credentials mode,
     // destination (CSP, TODO).
 
-    // Step 1. Let options be a script fetch options. TODO: credentials mode.
+    // Step 1. Let options be a script fetch options.
+    // We currently don't track credentials in our ScriptFetchOptions
+    // implementation. We translate this instead to cors.
+    CORSMode cors = mWorkerRef->Private()->WorkerCredentials() ==
+                            RequestCredentials::Include
+                        ? CORSMode::CORS_USE_CREDENTIALS
+                        : CORSMode::CORS_ANONYMOUS;
+
     RefPtr<ScriptFetchOptions> fetchOptions =
-        new ScriptFetchOptions(CORSMode::CORS_NONE, referrerPolicy, nullptr);
+        new ScriptFetchOptions(cors, referrerPolicy, nullptr);
 
     RefPtr<WorkerModuleLoader::ModuleLoaderBase> moduleLoader =
         static_cast<WorkerGlobalScopeBase*>(GetGlobal())->GetModuleLoader();